src/CMakeLists.txt

```diff
@@ -0,0 +1,47 @@
+# libtool versioning
+# See http://sources.redhat.com/autobook/autobook/autobook_91.html#SEC91 for details
+
+## increment if the interface has additions, changes, removals.
+set(LT_CURRENT 21)
+
+## increment any time the source changes; set 0 to if you increment CURRENT
+set(LT_REVISION 0)
+
+## increment if any interfaces have been added; set to 0
+## if any interfaces have been changed or removed. removal has
+## precedence over adding, so set to 0 if both happened.
+set(LT_AGE 13)
+
+math(EXPR LT_CURRENT_MINUS_AGE "${LT_CURRENT} - ${LT_AGE}")
+
+set(SONAME ${LT_CURRENT_MINUS_AGE}.${LT_AGE}.${LT_REVISION})
+set(SOVERSION ${LT_CURRENT_MINUS_AGE})
+
+if (ENABLE_CCE)
+    add_subdirectory("CCE")
+endif()
+add_subdirectory("common")
+add_subdirectory("CPE")
+add_subdirectory("CVE")
+add_subdirectory("CVSS")
+add_subdirectory("DS")
+add_subdirectory("OVAL")
+add_subdirectory("source")
+add_subdirectory("XCCDF")
+add_subdirectory("XCCDF_POLICY")
+
+set(OBJECTS_TO_LINK_AGAINST
+    $<TARGET_OBJECTS:common_object> $<TARGET_OBJECTS:cpe_object> $<TARGET_OBJECTS:cve_object> $<TARGET_OBJECTS:cvss_object>
+    $<TARGET_OBJECTS:ds_object> $<TARGET_OBJECTS:oval_object> $<TARGET_OBJECTS:ovaladt_object> $<TARGET_OBJECTS:ovalcmp_object>
+    $<TARGET_OBJECTS:ovalresults_object> $<TARGET_OBJECTS:oscapsource_object> $<TARGET_OBJECTS:rbt_object>
+    $<TARGET_OBJECTS:seap_object> $<TARGET_OBJECTS:xccdf_object> $<TARGET_OBJECTS:xccdfPolicy_object>
+)
+if (ENABLE_CCE)
+    list(APPEND OBJECTS_TO_LINK_AGAINST $<TARGET_OBJECTS:cce_object>)
+endif()
+
+add_library(openscap SHARED ${OBJECTS_TO_LINK_AGAINST})
+set_target_properties(openscap PROPERTIES VERSION ${SONAME} SOVERSION ${SOVERSION})
+
+install(TARGETS openscap
+		LIBRARY DESTINATION ${CMAKE_INSTALL_PREFIX}/lib)
```

Comment:
@mpreisler I am not sure about correct location to place the `libopenscap.so` library. Shouldn't this be `LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}` ?
On my system `${CMAKE_INSTALL_LIBDIR}` that leads to `${prefix}/lib64`, but our current buildsystem installs `libopenscap.so` to `${prefix}/lib` when I run `make install`. I would guess the `${prefix}/lib64` is more correct.

Reply:
Maybe we don't have to be so much compatible with results produced by automake ?

Reply:
@mpreisler What do you think about this problem?

Reply:
LIBDIR is definitely the way to go, hardcoded paths are almost always bad.

Reply:
@mpreisler Thank you for your reply. I agree. I will change it and other occurrences to `${CMAKE_INSTALL_LIBDIR}`.